Create LangChain_Chatbot

Anshhh369 opened the pull request:
The art and science of prompt engineering and its application in building LangChain chatbots. Equip with the skills to design and deploy intelligent LangChain chatbots tailored to diverse user needs and scenarios.

## PR Title
Outlining the significance of appointment scheduling optimization and the role of prompt engineering and AI.

## Description
Exploring the development and optimization of LangChain chatbots for appointment scheduling.

### Sections
Introduction of LangChain, Methodologies employed in the research process, analysis, conclusion, future direction.

### Screenshots
![1*WymmBqWHhmyGu_IJwK2qAg](https://github.com/parthasarathydNU/humanitarians-OPT-fork/assets/123016201/eaa85629-3fe7-40ba-8e3b-4e7b334bba81)


### Issues Faced
The Unpredictability of Models: Regardless of the model chosen, occasional instability in the outputs is encountered. Even with the adoption of the latest and best model available, such as OpenAI’s chat model, unpredictability persisted. It is a humbling realization that machine learning models have inherent variability. To address this, a robust testing is implemented and the model is continuously refined to achieve more consistent and reliable results.

### New Resources Added
The code is yet to be updated due to certain changes in integration tool. The intent is to upload the code file as soon as everything is working fine .


### How to Provide Public Access Links
Github link will be shared publicly with all the required materials including Jupyter notebook and a readme file.

parthasarathydNU commented:
Add a gitignore file to your project and add the .iml .xml files to it.

Only push markdown or other relevant files to the project.

divyajangid2496 commented:
Removed the binary files and added .xml and .iml to .gitignore file. Working on updating the README.md

divyajangid2496 commented:
Updated README file. Please let me know if it looks good.

parthasarathydNU commented on PROMPT_ENGINEERING_AND_AI/caseStudies/Medical_Diagnosis_With_Prompt_Engineering/Readme.md:
```diff
@@ -1,5 +1,8 @@
 # Revolutionizing Medical Diagnosis with AI and Prompt Engineering
 
+#### Document URL: https://docs.google.com/document/d/1a5DN8nwZb65hghlm54DG3givDTTN4vaVUqt39UQz6K0/edit
+
```
Your abstract and introduction present a compelling vision for the integration of prompt engineering and AI in medical diagnosis. The emphasis on interdisciplinary collaboration, patient care, and addressing challenges like privacy and accountability is particularly noteworthy. These sections effectively set the stage for a comprehensive exploration of the transformative potential of these technologies in healthcare.


## Recommendations for Improvement:

**Abstract Refinement:** Incorporate a sentence or two about the methodology or specific examples/case studies that the paper will delve into. This will help readers understand the scope and the approach you're taking in exploring the integration of prompt engineering and AI in medical diagnostics.

**Introduction Enhancements:** The introduction does an excellent job of presenting the problem statement and showcasing existing applications of AI in healthcare. To further strengthen this section, consider adding a paragraph that directly connects these examples to the role of prompt engineering. Highlighting how prompt engineering specifically contributes to the achievements of AI in these instances can offer a more nuanced understanding of its value.

**Citations and Evidence:** The citation provided at the end is a good start. Ensure that all statements, especially those involving statistics, innovations, and quotations, are backed by appropriate references throughout the paper. This will enhance the credibility and scholarly value of your case study.

divyajangid2496 replied:
Improved below points:
1. Abstract: "Through a comprehensive review of existing research literature such as PubMed, ...." 2nd Para, Line 2
2. Introduction: "Prompt engineering plays a crucial role in enabling the achievements of AI in these instances....." 2nd Last Paragraph newly added. Also, added some more data by adding paragraph 4.
3. Citation Improved

parthasarathydNU commented on PROMPT_ENGINEERING_AND_AI/caseStudies/Medical_Diagnosis_With_Prompt_Engineering/Readme.md:
```diff
@@ -24,21 +27,21 @@ This case study explores the transformative potential of artificial intelligence
 ### Introduction
-Introduces the problem statement and the role of AI in advancing medical diagnostic processes.
+In the landscape of healthcare, the integration of advanced technologies like artificial intelligence (AI) is reshaping medical diagnosis. With diagnostic errors accounting for a significant portion of medical mishaps and associated mortality rates, there is an urgent need for more efficient and precise diagnostic methods.By addressing challenges related to diagnostic accuracy, efficiency, and pa ...
 
```
### Feedback on Selected Aspects:

#### Introduction

Your introduction effectively sets the stage for the case study, highlighting the transformative potential of integrating prompt engineering and AI in medical diagnosis. The emphasis on enhancing diagnostic accuracy, efficiency, and patient outcomes is compelling and clearly communicated. The mention of exploring real-world applications and interdisciplinary collaborations is particularly strong, suggesting a comprehensive and collaborative approach to the study.

**Recommendations:**
- To further strengthen the introduction, consider providing specific examples of how prompt engineering has been applied in healthcare settings or mentioning ongoing research projects. This would offer readers concrete instances of the technology's impact.
- Clarify the role of prompt engineering in contrast to broader AI applications in healthcare. This distinction can help readers understand the unique contributions of prompt engineering to medical diagnostics.

#### Problem Statements

The problem statements section does an excellent job of identifying the challenges faced in medical diagnostics, such as diagnostic errors and the need for timely detection and interpretation of complex medical data. The urgency of addressing these issues is well-articulated, laying a solid foundation for the study's relevance and necessity.

**Recommendations:**
- While the section effectively outlines the challenges, including a brief discussion of why current diagnostic methods fall short and how AI and prompt engineering specifically address these shortcomings would enhance the narrative.
- Mentioning statistical data or studies to back up the claims about diagnostic errors and their consequences could add credibility and depth to the problem statements.

#### Case Study Sections

##### Abstract

The abstract succinctly captures the essence of the case study, emphasizing the synergy between prompt engineering and AI as a solution to current challenges in medical diagnosis. The focus on interdisciplinary collaboration, patient care, and technological advancements is well-placed.

**Recommendations:**
- It could be beneficial to briefly mention the case study's methodology or the types of data analysis that support your findings. This would give readers a clearer idea of the study's structure and approach.
- Highlighting a specific example or outcome from the case study in the abstract could make it even more engaging and informative.

##### Introduction (Section)

This introduction section reiterates the significance of AI in transforming medical diagnosis and the urgency of developing more efficient diagnostic methods. The commitment to addressing these challenges through AI-driven solutions and ethical considerations is clear and well-presented.

**Recommendations:**
- To avoid repetition with the main introduction, this section could further elaborate on how the study will explore these AI-driven solutions, possibly outlining the scope or specific areas of focus within medical diagnosis.
- A brief mention of the challenges or limitations of existing AI applications in healthcare and how prompt engineering can overcome these could provide a more nuanced view of the study's objectives.

Overall, your case study's selected aspects are well-crafted, clearly conveying the importance and potential impact of integrating AI and prompt engineering in medical diagnostics. By considering the recommendations provided, you can enhance the depth, clarity, and persuasiveness of your narrative, further emphasizing the significance of your research.

divyajangid2496 replied:
updated below sections of the README.md file as suggested:
Introduction
Problem Statement
Abstract
Introduction

parthasarathydNU reviewed (approved):
Here is my feedback on your work - feel free to incorporate this in your next PR.

parthasarathydNU reviewed (approved):
Good work on your initial draft - 

Changes in the next iteration: 

- Add visual aids to ease understanding of ideas and interesting facts that you identified through your work
- Update documentation of the Repo Readme and the Project Readme to submit to professor

Anshhh369 committed:
Create LangChain_Chatbot

The art and science of prompt engineering and its application in building LangChain chatbots. Equip with the skills to design and deploy intelligent LangChain chatbots tailored to diverse user needs and scenarios.

Anshhh369 committed:
Merge branch 'main' into patch-1

parthasarathydNU reviewed (changes requested):
Kindly update PR description to reflect changes accurately

parthasarathydNU reviewed (approved):
Let's keep this project on hold and look at other apps that you can build using RAG and later integrate Langchain on top of it. 

The next idea to work on: 

Build a RAG application where students can ask questions about a particular subject and get answers from it : 
1. What is the syllabus like
2. How chill is this course
3. What are the pre requisites to be able to take this course
4. What will I learn from this course
5. What are some projects that students have done as part of this coursework
6. Is this course work relevant if I want to get a job as a software enginer
7. .... think about what questions you want to be answered
8. How strict is this professor 

https://learn.deeplearning.ai/
https://www.youtube.com/watch?v=cVA1RPsGQcw
